test/yamlParser.test.ts

```diff
@@ -243,6 +244,65 @@ describe('YAML parser', () => {
       );
       assert(parsedDocument.documents[0].errors.length === 0, JSON.stringify(parsedDocument.documents[0].errors));
     });
+
+    it('parse aliases up to a depth', () => {
+      // If maxRefCount is set to 1, it will only resolve one layer of aliases, which means
+      // `b` below will inherit `a`, but `c` will not inherit `a`.
+      aliasDepth.maxRefCount = 1;
+      const parsedDocument = parse(`
+a: &a
+  foo: "web"
+b: &b
+  <<: *a
+c: &c
+  <<: *b
+`);
+      aliasDepth.maxRefCount = 1000;
```

Comment:
If `parse` crashes (due to bad code), `maxRefCount` will globally be left at `1`, which would cause other tests to fail perhaps improperly.

This may be safer:

```suggestion
      try {
        aliasDepth.maxRefCount = 1;
        const parsedDocument = parse(`
a: &a
  foo: "web"
b: &b
  <<: *a
c: &c
  <<: *b
`);
      } finally {
        aliasDepth.maxRefCount = 1000;
      }
```

Reply:
I very rarely use `finally` so I checked that this actually works as intended in a REPL:

```js
const x = () => {
    console.log("before")
    try {
        console.log("set 1")
        throw new Error("thrown by q");
    } finally {
        console.log("unset")
    }
    console.log("after")
}

x()
```

produces

```
before
set 1
unset
Uncaught Error: thrown by q
    at x (<anonymous>:5:15)
    at <anonymous>:1:1
```

Reply:
applied manually, had a syntax error